pkg/volume/git_repo/git_repo.go

```diff
@@ -126,10 +126,8 @@ func (b *gitRepoVolumeBuilder) SetUp() error {
 	return b.SetUpAt(b.GetPath())
 }
 
-// This is the spec for the volume that this plugin wraps.
-var wrappedVolumeSpec = &volume.Spec{
-	Volume: &api.Volume{VolumeSource: api.VolumeSource{EmptyDir: &api.EmptyDirVolumeSource{}}},
-}
+// This is the name for the volume that this plugin wraps, will be generated with random value in SetUpAt(..).
+var wrapperVolumeName string
```

Comment:
I didn't want to use volume name on the off chance that there was ever a
mode of one volume that wraps another mode of the same volume, or that
anyone might make a list of all volume names and get a conflict.  Using an
"illegal" value means we control the prefix.

On Fri, Dec 11, 2015 at 10:25 AM, Paul Morie <notifications@github.com>
wrote:

> In pkg/volume/git_repo/git_repo.go
> <https://github.com/kubernetes/kubernetes/pull/18445#discussion_r47386661>
> :
>
> > @@ -126,10 +126,8 @@ func (b *gitRepoVolumeBuilder) SetUp() error {
> >  	return b.SetUpAt(b.GetPath())
> >  }
> >
> > -// This is the spec for the volume that this plugin wraps.
> > -var wrappedVolumeSpec = &volume.Spec{
> > -	Volume: &api.Volume{VolumeSource: api.VolumeSource{EmptyDir: &api.EmptyDirVolumeSource{}}},
> > -}
> > +// This is the name for the volume that this plugin wraps, will be generated with random value in SetUpAt(..).
> > +var wrapperVolumeName string
>
> I guess I'm wondering, why not just use the volume name? Also, emptyDir
> should add a validation that the volume name is set.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/18445/files#r47386661>.
>